src/Microsoft.AspNetCore.Routing/DefaultInlineConstraintResolver.cs

```diff
@@ -82,7 +82,16 @@ private static IRouteConstraint CreateConstraint(Type constraintType, string arg
             // No arguments - call the default constructor
             if (argumentString == null)
             {
-                return (IRouteConstraint)Activator.CreateInstance(constraintType);
+                try
+                {
+                    return (IRouteConstraint)Activator.CreateInstance(constraintType);
+                }
+                catch (Exception exception)
+                {
+                    throw new InvalidRouteException(
+                        $"An error occurred while trying to create an instance of route constraint '{constraintType.FullName}'.",
+                        exception);
+                }
```

Comment:
Assuming we're OK with this approach, we probably want to throw this in place of a few other exceptions here, and also want to surround the calls to user code